flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableCatalog.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import org.apache.flink.table.catalog.exceptions.DatabaseNotExistException;
+import org.apache.flink.table.catalog.exceptions.TableNotExistException;
+
+import java.util.List;
+
+/**
+ * This interface is responsible for reading database/table/views/UDFs from a registered catalog.
+ * It connects a registered catalog and Flink's Table API.
+ */
+public interface ReadableCatalog {
```

Comment:
All methods in this class can throw a generic `CatalogException` if the communication fails or the API is not used as intended.

Reply:
Hi Timo, CatalogException you mentioned sounds like a RuntimeException that don't need to be declared. However, I noticed that the currently declared exceptions inherited also RuntimeException, which should be just exception. I'd suggest we change those as checked exception and reserve the generic CatalogException as a RuntimeException that might be thrown in all catalog APIs. How does this sound to you?

Reply:
Yes, checked exceptions make sense for all catalog methods. I'm fine with keeping `CatalogException` a runtime exception but I would add it to all catalog APIs to remind implementers that communication to catalogs is always error-prone.

Reply:
Sounds good. I will have CatalogException declared in those APIs.